addons/website/models/website_page.py

```diff
@@ -135,6 +136,7 @@ def clone_page(self, page_id, page_name=None, clone_menu=True):
             copy_param['url'] = self.env['website'].get_unique_path(url)
 
         new_page = page.copy(copy_param)
+        new_page.key = new_page.key + '_%s' % str(uuid.uuid4())[:6]
```

Comment:
Information missing: this is duplicate code of what the `copy` is supposed to be doing but https://github.com/odoo/odoo/commit/4ac2702c31f0e95f33f9ad554e7350bef9dab8bd#diff-dfebe5a93e1b8880e88268b024be4c6f106d144b20298d7bb6c4ae09a18bafd0R548

So this cannot be the right fix but not sure what the right one would be. Indeed, I would need @tivisse 's opinion :pray:

Reply:
@qsm-odoo You're right. When the view is copied in copy_data(), the view_id is updated in the page model but not the key. I've corrected the issue. We'll see if @tivisse agree with this.
https://github.com/odoo/odoo/pull/166351/files#diff-77bd6b19c39e211959885024bcad914655ff84cfc10c16633687d014e50aa69aR121-R123

Reply:
Well spotted, the fix seems correct to me :muscle: 

Thanks !

Reply:
> You're right. When the view is copied in copy_data(), the view_id is updated in the page record but not the key.

1. I don't understand that: website.page does not have a `key` field, it is just "inherit**s** from the view, referenced through the `view_id`. Why would it be necessary to force the key on the view itself (knowing it will just write on `view_id.key`) since we already force the view_id, referencing the view that contains that same key ?

2. Side point: In fact, I don't understand the whole copy override done here (which was there before the YTI's refactoring):
```python
result = super().copy_stuff()
if not default:
    return result
do_specific_stuff()
return result
```
`default` is documented as "field values to override in the original values of the copied record"... why would its presence or not be a condition to `do_specific_stuff` ? e.g. why do we have to copy the internal view by hand only when some default values are provided? :thinking: I am actually wondering why we have to copy by hand in any case, should not the ORM do it automatically know `view_id` is the inherits field?

Reply:
@qsm-odoo @tivisse 
![image](https://github.com/odoo/odoo/assets/22252071/8bcf0bc2-afc7-4270-aae9-6854bb38caa0)
Maybe related to that. I'm still investigating.

Reply:
@qsm-odoo @tivisse It seems more correct like this. https://github.com/odoo/odoo/pull/166351/files#diff-77bd6b19c39e211959885024bcad914655ff84cfc10c16633687d014e50aa69aR115-R122
Calling super() after the do_specific_stuff(). Otherwise the vals_list contains the 'key' value of the record copied, and thus must be updated.

Reply:
> 1. I don't understand that: website.page does not have a `key` field, it is just "inherit**s** from the view, referenced through the `view_id`. Why would it be necessary to force the key on the page itself (knowing it will just write on `view_id.key`) since we already force the view_id, referencing the view that contains that same key ?

This is what I understand from reading the code in [models.py](https://github.com/odoo/odoo/blob/saas-17.2/odoo/models.py):
While the key isn't "stored" on the `website.page` database table, it is still part of the `website.page` `_fields` dict, which flags it as a value to be copied over to the new record, only a couple of fields are not copied (create_date, write_date, etc.) including view_id.

However, since view_id is marked as inherited, it will either:
- Create (and not duplicate) a new view and insert the values from the previous record.
- If a view_id is defined in the vals (so if it was added in copy_data), it will update the record with the new vals.

That's where the issue is really, because the copy creates a new view, it does not go through "copy_data" and therefore does not create a new key. Even if we did duplicate the view, the key would still be overriden, as the key is copied over from the previous website.page

> `default` is documented as "field values to override in the original values of the copied record"... why would its presence or not be a condition to `do_specific_stuff` ? e.g. why do we have to copy the internal view by hand only when some default values are provided?

I do not have an answer for that and I don't think that this check is necessary. I assume it never caused any issue because we copy pages almost always with a default website_id.

> I am actually wondering why we have to copy by hand in any case, should not the ORM do it automatically knowing `view_id` is the inherits field?

We actually do not need to do it by hand, if we remove the code, the views are properly duplicated, but because `create` is used (with all the values inherited by `website.page`) on the views instead of `copy`, it would not fix the issue.

I'm sorry if that doesn't make a lot of sense or is poorly explained, I'm still not 100% sure how the ORM works.

IMO the best approach is still the original one, just duplicate the view and replace the `key` in the vals. Or do not duplicate the view and set a unique key in the vals manually. Using the defaults does not work as we can copy more than one record, and defaults are shared amongst all record (see comment below).

What do you think @qsm-odoo @SergeBayet @tivisse ?

CC @rdeodoo as I showed you this problem and you initially validated the fix of just changing the key.

Reply:
@detrouxdev As we discussed after delving in the ORM, it seems to me that the first fix was the best approach. In this case, we can go with the PR https://github.com/odoo/odoo/pull/163989 which uses this first approach and adds some tests.

Reply:
The other fix at the time we discussed it live with @detrouxdev seemed to be the good one, I did not dig into detail since then.
On top of that it's a one line fix + tests :+1: 
I believe you are delegated on both PR @detrouxdev so feel free to act accordingly :-)